netket/nn/module.py

```diff
@@ -0,0 +1,49 @@
+import flax
+from flax import linen as nn
+from flax import struct
+import jax
+from jax import numpy as jnp
+
+from netket.hilbert import AbstractHilbert
+from typing import Union, Tuple, Optional, Any, Callable
+
+from jax.random import PRNGKey
+
+
+class Module(nn.Module):
+    hilbert: AbstractHilbert = struct.field(pytree_node=False)
```

Comment:
Is hilbert actually still used anywhere?

The only place I could find was https://github.com/netket/netket/blob/7bc97aef015aef7e800c9e7d627bdabcf39c9e06/netket/models/mps.py#L39 however that is not even inheriting from this class.

Reply:
no it's not used.
It was an experiment I did at the beginning of a module more like a machine, with a fixed size (because you have the hilbert).

But then I changed idea because if I relied on the fact that modules has hilbert inside, then you cannot easily copy paste flax networks from the internet anymore.

Though I still find the idea of having a simple way to initialise the network not with a 'fake' input but with only the hilbert quite nice.

Reply:
I should delete this, or update it.